app/controllers/items_controller.rb

```diff
@@ -609,7 +609,12 @@ def add_workflow
       render :status => 500, :text => "#{params[:wf]} already exists!"
       return
     end
-    @object.initialize_workflow(params[:wf])
+    @object.create_workflow(params[:wf])
+
+    # sync up the workflows datastream with workflow service and force a reindex before redirection
+    reindex Dor::Item.find(params[:id])
```

Comment:
See my comment re: testing here: https://github.com/sul-dlss/argo/pull/113 `create_workflow` is unit tested in `dor-services` but the `reindex` call isn't.